Mage.Sets/src/mage/cards/f/FangsOfKalonia.java

```diff
@@ -0,0 +1,138 @@
+package mage.cards.f;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.keyword.OverloadAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.counters.CounterType;
+import mage.filter.StaticFilters;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.target.common.TargetControlledCreaturePermanent;
+
+/**
+ *
+ * @author grimreap124
+ */
+public final class FangsOfKalonia extends CardImpl {
+
+    public FangsOfKalonia(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[] { CardType.SORCERY }, "{1}{G}");
+
+        // Put a +1/+1 counter on target creature you control, then double the number of +1/+1 counters on each creature that had a +1/+1 counter put on it this way.
+        this.getSpellAbility().addEffect(new FangsOfKaloniaEffect());
+        this.getSpellAbility().addTarget(new TargetControlledCreaturePermanent());
+
+        // Overload {4}{G}{G}
+        this.addAbility(
+                new OverloadAbility(this, new FangsOfKaloniaOverloadEffect(), new ManaCostsImpl<>("{4}{G}{G}")));
+    }
+
+    private FangsOfKalonia(final FangsOfKalonia card) {
+        super(card);
+    }
+
+    @Override
+    public FangsOfKalonia copy() {
+        return new FangsOfKalonia(this);
+    }
+}
+
+class FangsOfKaloniaEffect extends OneShotEffect {
+
+    FangsOfKaloniaEffect() {
+        super(Outcome.Benefit);
+        staticText = "Put a +1/+1 counter on target creature you control, then double the number of +1/+1 counters on each creature that had a +1/+1 counter put on it this way";
+    }
+
+    private FangsOfKaloniaEffect(final FangsOfKaloniaEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        UUID target = source.getTargets().getFirstTarget();
+        Permanent preP1P1Creature = game.getPermanent(target);
+        if (preP1P1Creature != null) {
+            // Put a +1/+1 counter on target creature you control
+            int preP1P1Count = preP1P1Creature.getCounters(game).getCount(CounterType.P1P1);
+            preP1P1Creature.addCounters(CounterType.P1P1.createInstance(), source.getControllerId(), source, game);
+            game.applyEffects();
+            // then double the number of +1/+1 counters on each creature that had a +1/+1 counter put on it this way
+            Permanent postP1P1Creature = game.getPermanent(target);
+            if (postP1P1Creature != null) {
+                int postP1P1Count = postP1P1Creature.getCounters(game).getCount(CounterType.P1P1);
+                if (postP1P1Count > preP1P1Count) {
+
+                    int doubled = (postP1P1Count - preP1P1Count);
```

Comment:
1/ You double all +1/+1, not only those added previously by the effect
2/ Is the extra logging adding anything? There should already be logs on counters added.

Reply:
1/ Are you sure because I'm getting then number of counters then adding a counter then apply the difference in counters and it seemed to be working. Maybe I'm missing something?

preP1P1Count: 0
Add a counter
postP1P1Count: 1
postP1P1Count - PreP1P1Count = 1 - 0 = 1
Added 1 counter 

Of if you have [[Hardened Scales]]
preP1P1Count: 10
Add a counter
Hardened Scales makes it 2
postP1P1Count: 12
postP1P1Count - PreP1P1Count = 12 - 10 = 2
Added 2 counters
Hardened Scales makes it 3


2/ cleaned up the logging.

Reply:
I misphrased.

You should double all the P1P1 on the creatures, but here you only double the ones added by the effect.

Reply:
If a creature has 3 P1P1 already, casting Fangs on it should result in 8 counters.

Reply:
Yeah the weird wording of for it to work with the Overload.